src/identifiers.rs

```diff
@@ -215,19 +215,13 @@ impl RoomId {
     /// Parse/validate and create a new `RoomId`.
     #[napi(constructor, strict)]
     pub fn new(id: String) -> napi::Result<Self> {
-        Ok(Self::from(ruma::RoomId::parse(id).map_err(into_err)?))
-    }
-
-    /// Returns the user's localpart.
-    #[napi(getter)]
-    pub fn localpart(&self) -> String {
-        self.inner.localpart().to_owned()
-    }
-
-    /// Returns the server name of the room ID.
-    #[napi(getter)]
-    pub fn server_name(&self) -> ServerName {
-        ServerName { inner: self.inner.server_name().to_owned() }
+        let room_id = ruma::RoomId::parse(id).map_err(into_err)?;
+        match room_id.server_name() {
+            Some(_) => Ok(Self::from(room_id)),
+            None => Err(napi::Error::from_reason(
+                "Room ID does not have a valid server_name".to_owned(),
+            )),
+        }
```

Comment:
The previous version of this threw, as indicated by the tests. I don't think I want to align with the *wasm* bindings yet in favour of maintaining compatibility with previous versions of this library.

When we start moving towards WASM, I think this can be tidied up.